dgl_ptm/dgl_ptm/model/initialize_model.py

```diff
@@ -392,11 +394,27 @@ def step(self):
             #TODO add model dump here. Also check against previous save to avoid overwriting
             raise RuntimeError(f'execution of step failed for step {self.step_count}')
 
-    def run(self):
-        """ run the model for each step until the step_target is reached."""
+    def run(self, restart=False):
+        """
+        run the model for each step until the step_target is reached.
+
+        param: restart: boolean or int or a pair of ints, optional.
+        If True, the model is run from last checkpoint,
+        if an int, the model is run from the first milestone at that step,
+        if a pair of ints, the model is run from that milestone at that step.
+        Default False.
+        """
+
+        self.inputs = None
+        if isinstance(restart, bool):
+            if restart:
+                self.inputs = _load_model(f'./{self._model_identifier}')
+        elif isinstance(restart, int):
+            self.inputs = _load_model(f'./{self._model_identifier}/milestone_{restart}')
+        elif isinstance(restart, tuple):
+            self.inputs = _load_model(f'./{self._model_identifier}/milestone_{restart[0]}_{restart[1]}')
```

Comment:
I couldn't find where in the code the model state is saved in the directory `f'./{self._model_identifier}/milestone_{restart[0]}_{restart[1]}'`. This is also not tested.

Reply:
Based on our discussion, I will
* add an mkdir command to _save_model so we have control over where any (milestone) state save creates a new directory.
* add a test for the milestone_X_Y case

Reply:
As of commit e98f547bb0cc6846673ada97731d5134a431b2c7 this has been implemented.